docs/spec/relayer/Relayer.tla

```diff
@@ -267,478 +204,68 @@ PendingDatagrams(srcChainID, dstChainID) ==
 Relay(srcChainID, dstChainID) ==
-    LET datagramsAndRelayerUpdate == PendingDatagrams(srcChainID, dstChainID) IN
+    LET datagramsAndRelayerUpdate == ComputeDatagrams(srcChainID, dstChainID) IN
     /\ srcChainID /= dstChainID
     /\ outgoingDatagrams' = 
             [outgoingDatagrams EXCEPT 
                 ![dstChainID] = outgoingDatagrams[dstChainID] 
                                 \union 
                                 datagramsAndRelayerUpdate.datagrams
             ]
-    /\ relayerChainHeights' = datagramsAndRelayerUpdate.relayerUpdate       
-    /\ UNCHANGED chains
+    /\ relayerHeights' = datagramsAndRelayerUpdate.relayerUpdate       
+    /\ UNCHANGED <<chainAstore, chainBstore>>
 
-(***************************************************************************
- Component actions
- ***************************************************************************)
-\* Relayer
-\*    The relayer either
-\*        - updates its clients, or
-\*        - relays datagrams between two chains, or
-\*        - does nothing
-Relayer ==
-    \/ \E chainID \in ChainIDs : UpdateRelayerClients(chainID)
-    \/ \E srcChainID \in ChainIDs : \E dstChainID \in ChainIDs : Relay(srcChainID, dstChainID)
-    \/ UNCHANGED vars
+UpdateRelayer ==
+    \E chainID \in ChainIDs : UpdateRelayerClients(chainID)
     
-\* Environment
-\*    The environment takes the action Env!Next
-\*    and leaves the relayer variable unchanged
-Environment ==
-    /\ Env!Next
-    /\ UNCHANGED relayerChainHeights    
+CreateDatagrams ==
+    \E srcChainID \in ChainIDs : \E dstChainID \in ChainIDs : Relay(srcChainID, dstChainID)    
 
 (***************************************************************************
  Specification
  ***************************************************************************)
 \* Initial state predicate
-\*    Initially
-\*        - it is either the relayer's or the environment's turn
-\*        - the relayer chain heights are uninitialized 
-\*          (i.e., their height is nullHeight)
-\*        - the environment is initialized, by Env!Init    
+\*    Initially:
+\*        - the relayer heights are uninitialized (i.e., their height is nullHeight)
 Init == 
-    /\ turn \in {"relayer", "environment"}
-    /\ relayerChainHeights = [chainID \in ChainIDs |-> nullHeight]
-    /\ Env!Init    
+    /\ relayerHeights = [chainID \in ChainIDs |-> nullHeight]
+    /\ outgoingDatagrams = [chainID \in ChainIDs |-> {}]
     
 \* Next state action
-\*    The system consists of two components: relayer and environment.
-\*    In the system, the relayer and environment 
-\*    take alternate steps
+\*    The relayer either:
+\*        - updates its clients, or
+\*        - relays datagrams between two chains, or
+\*        - does nothing
 Next ==
-    \/ /\ turn = "relayer"
-       /\ Relayer
-       /\ turn' = "environment"
-    \/ /\ turn = "environment"
-       /\ Environment
-       /\ turn' = "relayer"     
+    \/ UpdateRelayer
```

Comment:
We probably don't need this (see comments above). We should eventually model client update as a consequence of trusted period expiration. But logic for it would probably also be part of CreateDatagrams.